exploration/text-vs-code.md

```diff
@@ -28,38 +32,69 @@ ICU MessageFormat and Fluent both support inline selectors
 separated from the text using `{…}` for multi-variant messages.
 ICU MessageFormat is the only known format that uses `{…}` to also delimit text.
 
-[Mustache templates](https://mustache.github.io/mustache.5.html)
-and related languages wrap "code" in `{{…}}`.
-In addition to placeholders that are replaced by their interpolated value during formatting,
-this also includes conditional blocks using `{{#…}}`/`{{/…}}` wrappers.
-
-[Handlebars](https://handlebarsjs.com/guide/) extends Mustache expressions
-with operators such as `{{#if …}}` and `{{#each …}}`,
-as well as custom formatting functions that become available as e.g. `{{bold …}}`.
-
-[Jinja templates](https://jinja.palletsprojects.com/en/3.1.x/templates/) separate
-`{% statements %}` and `{{ expressions }}` from the base text.
-The former may define tests that determine the inclusion of subsequent text blocks in the output.
-
-A cost that the message formatting and templating languages mentioned above need to rely on
-is some rule or behaviour that governs how to deal with whitespace at the beginning and end of a pattern,
-as statements may be separated from each other by newlines or other constructs for legibility.
+Formatting and templating are distinct operations with similarities.
+Both interpolate strings by using input values,
+provided as inputs alongisde the formatting pattern string or template,
+to produce a new string.
+Formatting usually refers to smaller strings, usually no larger than a sentence,
+whereas templating are used to produce larger strings, usually for text files of various file formats, often for HTML documents.
+
+There are two different styles of templating library design.
+Some languages/libraries enable the interopolation of the template substrings through programmatic expressions in "code mode" that print expressions to the output stream 
+(ex: [PHP](https://www.php.net/),
+[Freemarker](https://freemarker.apache.org/index.html)):
+```php
+<html>
+...
+	<?php
+		if (true) {
+			echo '<p>Hello World</p>';
```

Comment:
Just a nit, but I think both PHP and Freemarker would actually prefer the second method:

```php
<?php if (true): ?>
    <p>Hello, world!</p>
<?php endif ?>
```

Reply:
I didn't even know that is possible.

If I check https://www.w3schools.com/php/php_if_else.asp and https://www.php.net/manual/en/control-structures.if.php the option is not even mentioned.

It is mentioned in the "User Contributed Notes" of the manual.

But I kind of doubt that something that is not even mentioned in the official manual is the preferred way.

If anything this makes the point that having more than one way to do things, some more recommended than others, is not a good thing.

One will argue that PHP is not the best designed language. And it is not.
But do we want MF2 to become that? :-)